### DISCUSSION TITLE
How do you deal with missing or incomplete datasets in computer vision?

### DISCUSSION THREAD

**Discussion by Igor335566** [ultralytics/ultralytics#22548] [General]
Hey everyone!
I’m curious how people here handle dataset shortages for object detection / segmentation projects (YOLO, Mask R-CNN, etc.).

A few quick questions:

1. How often do you run into a lack of good labeled data for your models?

2. What do you usually do when there’s no dataset that fits — collect real data, label manually, or use synthetic/simulated data?

3. Have you ever tried generating synthetic data (Unity, Unreal, etc.) — did it actually help?

Would love to hear how different teams or researchers deal with this.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Igor335566, thank you for starting this discussion 🚀! This is an automated response and an Ultralytics engineer will also assist soon. In the meantime, we recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```

## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Great questions, Igor—data scarcity is common; our go‑to is: fine‑tune a pretrained YOLO11 on a small real “seed” set, use strong augmentations, then add targeted synthetic data for rare/unsafe cases while keeping validation/test strictly real to track sim‑to‑real gap; you can mix real+synthetic like this and tune a few hyps: 

```yaml
# dataset.yaml
train:
  - data/real/images/train
  - data/synth/images/train
val: data/real/images/val
names: [class0, class1]  # edit to your classes
```

```bash
yolo train model=yolo11n.pt data=dataset.yaml epochs=100 imgsz=640 mosaic=1.0 mixup=0.15 copy_paste=0.1
```

If you generate with Unity/Unreal, prioritize domain randomization over photorealism and ensure perfect labels; for context see the synthetic data overview and sim‑to‑real tips in our articles on [Synthetic Data](https://www.ultralytics.com/glossary/synthetic-data), the [YOLO data augmentation guide](https://docs.ultralytics.com/guides/yolo-data-augmentation/), and our [Data collection and annotation strategies](https://docs.ultralytics.com/guides/data-collection-and-annotation/).

**Reply by Igor335566** [reply]
Thank you very much for the answer!

**Reply by glenn-jocher** [reply]
You’re welcome—if you end up trying a mixed real + synthetic setup, feel free to share a brief summary of your pipeline and results here, as those practical examples are very helpful for others.
